progressReport/app.py

```diff
@@ -65,7 +66,30 @@
 
 app = Flask(__name__)
 
-@app.route('/', methods=["GET"])
+"""
+Validates the fields of the mastery learning post request
+"""
+def validate_mastery_learning_post_request(request_as_json):
+    schema = {
+        "type": "object",
+
+        "properties": {
+            # These fields can optionally be included in the POST request.
+            "school": {"type": "string"},
+            "class": {"type": "string"},
+            "class_mastery": {"type": "string"},
+        },
+        "patternProperties": {
+            "^(?!school$|class$|class_mastery).*": {"type": "integer"}  # Matches mastery fields (fields other than "school", "class", and "class_mastery")
+        },
+        "additionalProperties": False,  # Disallow undefined properties
+        "required": []  # No fields are required
+    }
+
+    # Intentionally not handling the error an improper format may produce for now.
+    jsonschema.validate(instance=request_as_json, schema=schema)
+
+@app.route('/', methods=["GET", "POST"])
```

Comment:
Perhaps we can change the specification of the POST request such that it contains all info we currently relay to CM in the url, including optional fields, so that we can use the index method to handle the POST request.

Reply:
Agreed.  Thses should be other fields in the body as well (post requests don't take in query params; in fact, GET request query params are functionally just encoded in the body of the request).